File: Python/7py 1/chess_board.py
"""
|--------------------------------------------------------------------------
| draw me a chessboard
|--------------------------------------------------------------------------
|
"""

def chess_board(rows, columns):
   res = []
   for i in range(rows):
      temp = []
      for j in range(columns):
         temp.append("O") if (i + j) % 2 == 0 else temp.append("X")
      res.append(temp)
   return res


def chess_board(rows, cols):
   return [list(('OX'*cols)[row%2:row%2+cols]) for row in range(rows)]


def chess_board(rows, columns):
   return [["OX"[(i+j)%2] for j in range(columns)] for i in range(rows)]


print(chess_board(6, 4))

File: PHP/7php 1/chessBoard.php
<?php
/*
|--------------------------------------------------------------------------
| draw me a chessboard
|--------------------------------------------------------------------------
|
*/

function chessBoard($rows, $columns) {
   $res = [];
   for ($i = 0; $i < $rows; ++$i) {
      $temp = [];
      for ($j = 0; $j < $columns; ++$j) {
         ($i + $j) % 2 == 0 ? array_push($temp, "O") : array_push($temp, "X");
      }
      array_push($res, $temp);
   }
   return $res;
}


function chessBoard($rows, $columns) {
   $res = [];
   for ($i = 0; $i < $rows; $i++) array_push($res, array_map(function ($i) {
      return $i % 2 ? "X" : "O";
   }, range($i, $i + $columns - 1)));
   return $res;
}


function chessBoard($rows, $columns) {
   $res = [];
   for ($row = 0; $row < $rows; $row++) {
      for ($column = 0; $column < $columns; $column++) {
         $res[$row][$column] = ($row + $column) % 2 ? "X" : "O";
      }
   }
   return $res;
}


print_r(chessBoard(6, 4));